### ISSUE TITLE
Deploy on standard k8s environments

### ISSUE THREAD

**Issue by jxadro** [kiali/kiali.github.io#13]
Hi.

I have tried to deploy it in other k8s installation (IBM Cloud Private) and it works. I had to remove some proprietary configuraion for OpenShift:

I deployed everything on namespace istio-system

    Config Map:
    
    apiVersion: v1
    kind: ConfigMap
    metadata:
     name: kiali
     labels:
       app: kiali
       version: master
    data:
     config.yaml: |
       server:
         port: 20001
         static_content_root_directory: /opt/kiali/console
         credentials:
           username: admin
           password: admin
    
    Cluster role:
    
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
     name: kiali
     labels:
       app: kiali
       version: master
    rules:
    - apiGroups: ["","apps", "autoscaling"]
     attributeRestrictions: null
     resources:
     - configmaps
     - namespaces
     - nodes
     - pods
     - projects
     - services
     - endpoints
     - deployments
     - horizontalpodautoscalers
     verbs:
     - get
     - list
     - watch
    - apiGroups: ["config.istio.io"]
     attributeRestrictions: null
     resources:
     - routerules
     - destinationpolicies
     - rules
     - circonuses
     - deniers
     - fluentds
     - kubernetesenvs
     - listcheckers
     - memquotas
     - opas
     - prometheuses
     - rbacs
     - servicecontrols
     - solarwindses
     - stackdrivers
     - statsds
     - stdios
     - apikeys
     - authorizations
     - checknothings
     - kuberneteses
     - listentries
     - logentries
     - metrics
     - quotas
     - reportnothings
     - servicecontrolreports
     verbs:
     - get
     - list
     - watch
    - apiGroups: ["networking.istio.io"]
     attributeRestrictions: null
     resources:
     - virtualservices
     - destinationrules
     verbs:
     - get
     - list
     - watch
    
    
    ClusterRoleBinding:
    
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
     name: kiali
     labels:
       app: kiali
       version: master
    roleRef:
     apiGroup: rbac.authorization.k8s.io
     kind: ClusterRole
     name: kiali
    subjects:
    - kind: ServiceAccount
     name: kiali
     namespace: istio-system
    
    
    ServiceAccount, Service and Deployment:
    
    apiVersion: v1
    kind: ServiceAccount
    metadata:
     name: kiali
     labels:
       app: kiali
       version: master
    ---
    apiVersion: v1
    kind: Service
    metadata:
     name: kiali
     labels:
       app: kiali
       version: master
    spec:
     type: NodePort
     ports:
     - name: tcp
       protocol: TCP
       port: 20001
     selector:
       app: kiali
       version: master
    ---
    apiVersion: extensions/v1beta1
    kind: Deployment
    metadata:
     name: kiali
     labels:
       app: kiali
       version: master
    spec:
     replicas: 1
     selector:
       matchLabels:
         app: kiali
         version: master
     template:
       metadata:
         name: kiali
         labels:
           app: kiali
           version: master
       spec:
         serviceAccount: kiali
         containers:
         - image: kiali/kiali:latest
           name: kiali
           command:
           - "/opt/kiali/kiali"
           - "-config"
           - "/kiali-configuration/config.yaml"
           - "-v"
           - "4"
           env:
           - name: ACTIVE_NAMESPACE
             valueFrom:
               fieldRef:
                 fieldPath: metadata.namespace
           volumeMounts:
           - name: kiali-configuration
             mountPath: "/kiali-configuration"
         volumes:
         - name: kiali-configuration
           configMap:
             name: kiali

**Comment by pilhuhn** [comment]
This is great. 
Have you seen the work in github.Com/kiali/kiali/deploy/kubernetes? 

If yours is different, could you please open a pull-request on it? 

I wonder if you have experience with helm charts for deployment on Kubernetes and could help us there 

Thank you 
 - Heiko

**Comment by jmazzitelli** [comment]
Yes, this is great! As @pilhuhn  mentioned, @jxadro can you make your changes to our k8s config files and submit a PR - we'd be grateful to have a working k8s config. The two files are here:

https://github.com/kiali/kiali/tree/master/deploy/kubernetes

**Comment by jmazzitelli** [comment]
FWIW: I got the deploy/kubernetes/kiali.yaml to work. Just briefly looking, it appears the changes I made are the same as what is listed in this issue.

See: https://github.com/kiali/kiali/pull/202/files

I have it working on the latest minikube.

**Comment by pilhuhn** [comment]
@jxadro Is this now solved for you?
